### ISSUE TITLE
group edit error with cockpit 0.9.3

### ISSUE THREAD

**Issue by richardt3** [serjoscha87/cockpit_GROUPS#11]
Hi @serjoscha87 

I've installed a clean version of cockpit on a wampserver, php 7.3.12 and then installed Groups from the zip download

Creating a group looks fine, but editing the group is throwing an error. Not sure where to look.

Any thoughts?
Many thanks 

php_error.log

```[05-Dec-2019 13:57:00 UTC] PHP Warning:  Illegal string offset 'key' in C:\wamp64\cockpit\storage\tmp\group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 44
[05-Dec-2019 13:57:00 UTC] PHP Stack trace:
[05-Dec-2019 13:57:00 UTC] PHP   1. {main}() C:\wamp64\cockpit\index.php:0
[05-Dec-2019 13:57:00 UTC] PHP   2. LimeExtra\App->run() C:\wamp64\cockpit\index.php:48
[05-Dec-2019 13:57:00 UTC] PHP   3. LimeExtra\App->dispatch() C:\wamp64\cockpit\lib\Lime\App.php:983
[05-Dec-2019 13:57:00 UTC] PHP   4. LimeExtra\App->render_route() C:\wamp64\cockpit\lib\Lime\App.php:1036
[05-Dec-2019 13:57:00 UTC] PHP   5. LimeExtra\App->Lime\{closure:C:\wamp64\cockpit\lib\Lime\App.php:893-900}() C:\wamp64\cockpit\lib\Lime\App.php:1090
[05-Dec-2019 13:57:00 UTC] PHP   6. LimeExtra\App->invoke() C:\wamp64\cockpit\lib\Lime\App.php:899
[05-Dec-2019 13:57:00 UTC] PHP   7. Cockpit\Controller\Groups->group() C:\wamp64\cockpit\lib\Lime\App.php:1114
[05-Dec-2019 13:57:00 UTC] PHP   8. Cockpit\Controller\Groups->render() C:\wamp64\cockpit\addons\Groups\Controller\Groups.php:41
[05-Dec-2019 13:57:00 UTC] PHP   9. LimeExtra\App->view() C:\wamp64\cockpit\lib\LimeExtra\Controller.php:51
[05-Dec-2019 13:57:00 UTC] PHP  10. Lexy->file() C:\wamp64\cockpit\lib\LimeExtra\App.php:123
[05-Dec-2019 13:57:00 UTC] PHP  11. lexy_include_with_params() C:\wamp64\cockpit\lib\Lexy.php:132
[05-Dec-2019 13:57:00 UTC] PHP  12. include() C:\wamp64\cockpit\lib\Lexy.php:443 ```

**Comment by richardt3** [comment]
![edit](https://user-images.githubusercontent.com/26472369/70242392-a88be700-1768-11ea-8f4d-79c2e0d8df82.png)

**Comment by bassauer-storms-media** [comment]
Hey, thank you for your request.

I just updated my cockpit test-installation to the last version (running on ubuntu 18.04) and I don't have any issues - everything works just fine without any errors.

So I actually cannot help you :/

The stacks trace does not say anything at all :/

Perhaps it would help to see the full xdebug stack trace that is printed to the input fields in your case. Could you use your browser dev tool to increase height of the element, so we can see its content?

**Comment by richardt3** [comment]
```<br /> <font size='1'><table class='xdebug-error xe-warning' dir='ltr' border='1' cellspacing='0' cellpadding='1'> <tr><th align='left' bgcolor='#f57900' colspan=<input class="uk-width-1-4 uk-form-small" type="text" placeholder="value" value="<br /> <font size='1'><table class='xdebug-error xe-warning' dir='ltr' border='1' cellspacing='0' cellpadding='1'> <tr><th align='left' bgcolor='#f57900' colspan=" 5="" span="" style="background-color: #cc0000;color: #fce94f;font-size: small;" warning:="" illegal="" string="" offset="" val="" in="" c:="" wamp64="" cockpit="" storage="" tmp="" group="" php="" 2f2cd7b5eb54d252eb883c21a6019784="" lexy="" on="" line="" i="" 46="" call="" stack="" timememoryfunctionlocation="" 10="" 0004408464="" 7="" index="" b="" :="" 0="" 20="" 0173663872limeextra="" app-="" gt="" run="" 48="" 30="" 0173665136limeextra="" dispatch="" app="" 983="" 40="" 0175666448limeextra="" render_route="" 1036="" 50="" lime="" closure:c:="" lib="" php:893-900="" 1090="" 60="" 0175667288limeextra="" invoke="" 899="" 70="" 0176667416cockpit="" controller="" groups-="" 1114="" 80="" 0181673792cockpit="" render="" groups="" 41="" 90="" 0181673888limeextra="" view="" 51="" 100="" 0184675168lexy-="" file="" 123="" 110="" 0187691856lexy_include_with_params="" 132="" 120="" 0187693064include="" font="" color="#00bb00" 443="" div=""><input class="uk-width-1-4 uk-form-small" type="text" placeholder="value" value="<br /> <font size='1'><table class='xdebug-error xe-warning' dir='ltr' border='1' cellspacing='0' cellpadding='1'> <tr><th align='left' bgcolor='#f57900' colspan=" 5="" span="" style="background-color: #cc0000;color: #fce94f;font-size: small;" warning:="" illegal="" string="" offset="" val="" in="" c:="" wamp64="" cockpit="" storage="" tmp="" group="" php="" 2f2cd7b5eb54d252eb883c21a6019784="" lexy="" on="" line="" i="" 46="" call="" stack="" timememoryfunctionlocation="" 10="" 0004408464="" 7="" index="" b="" :="" 0="" 20="" 0173663872limeextra="" app-="" gt="" run="" 48="" 30="" 0173665136limeextra="" dispatch="" app="" 983="" 40="" 0175666448limeextra="" render_route="" 1036="" 50="" lime="" closure:c:="" lib="" php:893-900="" 1090="" 60="" 0175667288limeextra="" invoke="" 899="" 70="" 0176667416cockpit="" controller="" groups-="" 1114="" 80="" 0181673792cockpit="" render="" groups="" 41="" 90="" 0181673888limeextra="" view="" 51="" 100="" 0184675168lexy-="" file="" 123="" 110="" 0187691856lexy_include_with_params="" 132="" 120="" 0187693064include="" font="" color="#00bb00" 443="" div="">```

**Comment by richardt3** [comment]
Might be easier to read the screenshot.
![edit2](https://user-images.githubusercontent.com/26472369/70247382-3075ef00-1771-11ea-8756-3b601644b1af.png)

**Comment by richardt3** [comment]
Hi 

Thanks for your quick response :-)  This is running on windows server (sorry!)and under WampServer (latest). I'll spin up a linux server as well to see what I get. 

thanks Richard

**Comment by richardt3** [comment]
Hi again, 

I've just installed ubuntu 18 with php 7.3 and cockpit 0.9.3. connected to /install and created the cockpit database. No further configuration performed. Login as admin to prove it works.
Then added in the Groups Module
Select the Groups menu item and add a group. Vars are shown. 
Save and Reopen the interface is clean but the vars are empty
![edit-linux-version](https://user-images.githubusercontent.com/26472369/70253586-615b2180-177b-11ea-8579-0f1b4578e4ef.png)

php error log has:  

```Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'key' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 44
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'val' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 46
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'key' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 44
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'val' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 46
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'key' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 44
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'val' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 46
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'key' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 44
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'val' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 46
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'key' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 44
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'val' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 46
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'key' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 44
Dec  5 15:49:43 cockpit2 php: PHP Warning:  Illegal string offset 'val' in /var/www/html/storage/tmp/group.php.2f2cd7b5eb54d252eb883c21a6019784.lexy.php on line 46```

Feels like I'm hitting the same error.
Can you think of anything I'm doing wrong?

Thanks

**Comment by bassauer-storms-media** [comment]
Ah I see. Gonna check this coming week. Looks like this is going to be a minor / quick fix.

Thank you for your contribution!

**Comment by aMUSiC** [comment]
I am reproducing the same issue on ubuntu

**Comment by aolko** [comment]
so where's the fix?

**Comment by bassauer-storms-media** [comment]
there is no fix yet @aolko

**Comment by aolko** [comment]
ETA?

**Comment by bassauer-storms-media** [comment]
none, I got literally no time at the moment

should be a small thing to fix - could even done by someone else

sorry

**Comment by maettyhawk** [comment]
@aolko @aMusic @richardt3 
Here's my quick and dirty fix for that (replace group.php line 39-57 by the following):
<pre>
                                @if(isset($group['vars']))
                                    @foreach( $group['vars'] as $key => $val)
                                    <div class="uk-panel uk-panel-box uk-panel-card var-row uk-margin-small-bottom">
                                        <div class="uk-grid uk-grid-small">
                                            <div class="uk-flex-item-1 uk-flex">
                                                <input class="uk-width-1-4 uk-form-small" type="text" placeholder="key" value="{{$key}}">
                                                <i class="uk-width-1-4 uk-text-center uk-icon-arrows-h"></i>
                                                <input class="uk-width-1-4 uk-form-small" type="text" placeholder="value" value="{{$val}}">
                                                <div class="uk-width-1-4 uk-text-right">
                                                    <i class="uk-icon-trash" style="cursor: pointer" onclick="$(this).parents('.var-row').remove()"></i>
                                                </div>
                                            </div>
                                        </div>
                                    </div>
                                    @endforeach
                                @endif
</pre>

**Comment by serjoscha87** [comment]
I had some time. I fixed this issue with the last commit (695168b)

So now you guys are again able to edit the group vars / settings
